src/Servers/Kestrel/Core/src/Internal/Http2/Http2MessageBody.cs

```diff
@@ -14,6 +14,7 @@ public class Http2MessageBody : MessageBody
     {
         private readonly Http2Stream _context;
         private ReadResult _readResult;
+        private SequencePosition? _prevExamined;
```

Comment:
Looking at the other failure I'm handling in System.IO.Pipelines, I think this is fundamentally busted 😢 . It's impossible to use SequencePosition across read calls today.

Reply:
Any change we can make to SequencePosition to make this work?